linkerd/app/core/src/svc.rs

```diff
@@ -36,26 +33,22 @@ impl<L> Layers<L> {
         Layers(Pair::new(self.0, outer))
     }
 
-    /// Buffer requests when when the next layer is out of capacity.
-    pub fn push_pending(self) -> Layers<Pair<L, pending::Layer>> {
-        self.push(pending::layer())
+    /// Wraps an inner `MakeService` to be a `NewService`.
+    pub fn push_into_new_service(self) -> Layers<Pair<L, new_service::FromMakeServiceLayer>> {
+        self.push(new_service::FromMakeServiceLayer::default())
```

Comment:
bikeshed: IMO `pending` described the behavior of this thing a little clearer...yes, it turns a `MakeService` into a `NewService`, but that seems incidental to the underlying behavior of allowing a pending `Future<Service>` to be treated as a `Service`...I can learn to live with this terminology, though... not a blocker!

Reply:
> but that seems incidental to the underlying behavior of allowing a pending Future<Service> to be treated as a Service

That's not really what it does though? It explicitly modeled a MakeService as a Make (now, NewService). It's an adapter type.

"pending", to me, sounds much more about readiness